doc/EPIC.md

```diff
@@ -0,0 +1,110 @@
+# EPIC Design 
+
+This file introduces EPIC and documents the design rationales and best practices for EPIC-HP.
+
+## Introduction
+One important security property of SCION is that end hosts are not allowed to send packets along arbitrary paths, but only along paths that were advertised by all on-path ASes. 
+This property is called *path authorization*. 
+The ASes only advertise paths that serve their economic interests, and path authorization protects those routing decisions against malicious end hosts.
+
+In SCION, this is implemented by having the ASes create authenticators during beaconing, which end hosts then have to include in the MAC field of their packets. Those MAC fields prove to the ASes, that the packets are allowed to traverse them.
+The MAC fields are static, meaning that they are the same for every packet on that path.
+
+However, this implementation of path authorization is insufficient against more sophisticated adversaries that are able to derive MAC fields, for example by brute-forcing them or by observing them in the data plane packets. Because the MACs are static, once derived MACs for one packet can be reused by the adversary to send arbitrarily many other packets (until the authenticators expire).
+
+The EPIC (Every Packet Is Checked) protocol [[1]](#1) solves this problem by introducing per-packet MACs.
+Even if an adversary is able to brute-force the MACs for one packet, the MACs cannot be reused to send any other traffic.
+Observing MACs from packets in the data plane does also not help an attacker, as he still needs to derive new MACs for every packet he wants to send over an unauthorized path.
+This is especially important for hidden paths [[2]](#2). Hidden paths are paths which are not publicly announced, but only communicated to a group of authorized sources. If one of those sources sends traffic on the hidden path using SCION path type packets, an adversary can observe the MACs and reuse them to send traffic on the hidden path himself. This allows the adversary to reach services that were meant to be hidden, or to launch DOS-attacks directed towards them.
+EPIC precludes such attacks, making hidden paths more secure.
+
+## EPIC-HP Overview
+EPIC-HP (EPIC for Hidden Paths) provides the improved security properties of EPIC on the very last inter-AS link of a path. It is meant as a lightweight EPIC version and is specifically designed to better protect hidden paths.
+
+### Assumptions
+EPIC-HP makes the following assumptions necessary to provide a meaningful level of security:
+- The AS protected by the hidden path is the last AS.
+By "last AS" we mean that the beacon which defined the hidden path ends in the local interface of this AS. Or stated differently: the AS behind the hidden path does not forward the beacon defining the hidden path to further downstream/peering ASes. 
```

Comment:
I'm not sure "local interface" is standard terminology.

Reply:
I found the term in the border router code (/go/border/rpkt/route.go and /go/border/setup.go).